client/rest-high-level/src/test/java/org/elasticsearch/client/DataFrameTransformIT.java

```diff
@@ -186,9 +187,7 @@ public void testGetTransform() throws IOException {
         DataFrameTransformConfig transform = validDataFrameTransformConfig(id, sourceIndex, "pivot-dest");
 
         DataFrameClient client = highLevelClient().dataFrame();
-        AcknowledgedResponse ack = execute(new PutDataFrameTransformRequest(transform), client::putDataFrameTransform,
-                client::putDataFrameTransformAsync);
-        assertTrue(ack.isAcknowledged());
+        putTransform(transform);
```

Comment:
It feels that the paradigm we've followed with ML jobs where we have a cleaner class (see `MlTestStateCleaner`) might be nice to use here too. It frees up the tests from having to worry about cleaning up and thus makes the test easier to write/read and less error prone. Of course, we can push this fix as is and refactor in a different time & PR.

Reply:
`transformsToClean` was an expedient I added when first writing the tests because I hadn't implemented GET transforms at that point and needed a way to track the DFs that had been created. It probably shouldn't have survived the rewrites but it's not such a problem. Deleting the indices between tests removes the config and there is a call to `waitForPendingTasks` in the teardown which would catch any created DFs not tracked in `transformsToClean`. So a low priority refactor.